### ISSUE TITLE
Option to use compile rather than runtime scope

### ISSUE THREAD

**Issue by jglick** [johnpoth/jshell-maven-plugin#9]
Even setting aside #8, this mojo is currently useless for Jenkins plugins since it uses the runtime scope, which is _not_ generally a superset of compile scope: it lacks provided and system dependencies. In the case of a Jenkins plugin, `jenkins-core.jar` and its dependencies are `provided`, so half the classpath is missing.

My current workaround for both issues:

```bash
jshell --class-path $(mvn -Dsilent -q -Dmdep.outputFile=/dev/stdout -DincludeTypes=jar dependency:build-classpath):$(pwd)/target/classes:$(pwd)/target/test-classes
```

**Comment by johnpoth** [comment]
Hi @jglick, after checking, it seems that the test classpath does include `provided` and `system` dependencies. I just checked with the reproducer you provided in #8. If you would like to verify, you can clone this project and built it then run : 
`mvn com.github.johnpoth:jshell-maven-plugin:1.3-SNAPSHOT:run -DtestClasspath -X`

You see the classpath entries logged with `provided` and `system` dependencies included. Therefore I think we can close this issue. I'll release 1.3 sometime today I think.

Love you workaround btw !

**Comment by jglick** [comment]
Will check from a source build, since while it looks like you _tried_ to release 1.3, it is not out there:

```console
$ docker run --rm -ti maven mvn com.github.johnpoth:jshell-maven-plugin:1.2:run
[INFO] Scanning for projects...
Downloading from central: https://repo.maven.apache.org/maven2/com/github/johnpoth/jshell-maven-plugin/1.2/jshell-maven-plugin-1.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/com/github/johnpoth/jshell-maven-plugin/1.2/jshell-maven-plugin-1.2.pom (4.7 kB at 16 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/com/github/johnpoth/jshell-maven-plugin/1.2/jshell-maven-plugin-1.2.jar
Downloaded from central: https://repo.maven.apache.org/maven2/com/github/johnpoth/jshell-maven-plugin/1.2/jshell-maven-plugin-1.2.jar (9.2 kB at 167 kB/s)
[INFO] 
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.685 s
[INFO] Finished at: 2020-03-27T20:47:41Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.johnpoth:jshell-maven-plugin:1.2:run (default-cli): Goal requires a project to execute but there is no POM in this directory (/). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
$ docker run --rm -ti maven mvn com.github.johnpoth:jshell-maven-plugin:1.3:run
[INFO] Scanning for projects...
Downloading from central: https://repo.maven.apache.org/maven2/com/github/johnpoth/jshell-maven-plugin/1.3/jshell-maven-plugin-1.3.pom
[WARNING] The POM for com.github.johnpoth:jshell-maven-plugin:jar:1.3 is missing, no dependency information available
Downloading from central: https://repo.maven.apache.org/maven2/com/github/johnpoth/jshell-maven-plugin/1.3/jshell-maven-plugin-1.3.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.628 s
[INFO] Finished at: 2020-03-27T20:47:47Z
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin com.github.johnpoth:jshell-maven-plugin:1.3 or one of its dependencies could not be resolved: Could not find artifact com.github.johnpoth:jshell-maven-plugin:jar:1.3 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
```

**Comment by jglick** [comment]
The fix of #8 indeed helps, but this is still not an adequate solution due to #10.

**Comment by johnpoth** [comment]
Fixed in 1.3
